builtin/providers/aws/resource_aws_sns_topic_subscription.go

```diff
@@ -178,6 +197,13 @@ func subscribeToSNSTopic(d *schema.ResourceData, snsconn *sns.SNS) (output *sns.
 	protocol := d.Get("protocol").(string)
 	endpoint := d.Get("endpoint").(string)
 	topic_arn := d.Get("topic_arn").(string)
+	endpoint_auto_confirms := d.Get("endpoint_auto_confirms").(bool)
+	max_fetch_retries := d.Get("max_fetch_retries").(int)
+	fetch_retry_delay := time.Duration(d.Get("fetch_retry_delay").(int))
+
+	if strings.Contains(protocol, "http") && !endpoint_auto_confirms {
```

Comment:
Yeah, I see your point. This will be eventually possible after merging https://github.com/hashicorp/terraform/pull/3641 but I wouldn't want to block this PR on that.

The only suggestion I have regarding this is that we should rather be comparing protocols as IDs (i.e. `(protocol == "http" || protocol == "https")`), not arbitrary strings.

I almost thought on first sight you forgot about `https`.
 Also there might be a new service in the future which will actually contain `http` in the name and that would make things even more confusing.